invenio_app_rdm/theme/assets/js/react_invenio_deposit/DepositController.js

```diff
@@ -20,37 +24,48 @@ export class DepositController {
     console.log("Validate record", record);
   }
 
-  async save_draft(record) {
+  async save_draft(record, { formik, store }) {
     // Saves a draft of the record
-    let payload = record;
+    const recordSerializer = store.config.recordSerializer;
+    let payload = recordSerializer.deserialize(record);
```

Comment:
Hmm the terms `deserialize` and `serialize` are perhaps confusing. I associate serialization to transforming an input into a format that can be sent over the network: frontend memory representation to represenation the backend can deserialize. See https://stackoverflow.com/questions/633402/what-is-serialization . So I would have expected this to be `serialize`. If it's too overloaded, we can also use:

serialize === marshal === dump
deserialize === unmarshal === load  

And if we need to convert from a json representation to another one (say more compatible with the expectations of the front-end form library), well, `convertToOtherOne` where `convert` is the action word is probably best. Synonyms like `transform`, `represent`, `reshape`... are good too.

Reply:
I went for inverting `serialize`/`deserialize`. We can discuss in person if we need to rename the 2 methods :) Thanks for the detailed explanation on this :)